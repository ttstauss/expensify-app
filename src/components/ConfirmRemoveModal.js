import React from 'react'
import Modal from 'react-modal'

const ConfirmRemoveModal = props => (
  <Modal
    isOpen={props.modalIsOpen}
    contentLabel="Remove?"
    closeTimeoutMS={200}
    className="modal"
  >
    <h3 className="modal__title">Are you sure?</h3>
    <p className="modal__subtitle">You will not be able to recover this expense.</p>
    <div className="button-group">
      <button className="button button--secondary" onClick={props.handleCancelRemove}>Cancel</button>
      <button className="button button--warning" onClick={props.handleConfirmRemove}>Yes, delete it!</button>
    </div>
  </Modal>
)

if (process.env.NODE_ENV !== 'test') Modal.setAppElement('#app')

export default ConfirmRemoveModal