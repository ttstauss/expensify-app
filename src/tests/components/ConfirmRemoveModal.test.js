import React from 'react'
import { shallow } from 'enzyme'
import ConfirmRemoveModal from '../../components/ConfirmRemoveModal'

test('should render ConfirmRemoveModal correctly', () => {
  const wrapper = shallow(<ConfirmRemoveModal />)
  expect(wrapper).toMatchSnapshot()
})