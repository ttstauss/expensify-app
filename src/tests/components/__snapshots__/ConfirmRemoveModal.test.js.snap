// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ConfirmRemoveModal correctly 1`] = `
<Modal
  ariaHideApp={true}
  bodyOpenClassName="ReactModal__Body--open"
  className="modal"
  closeTimeoutMS={200}
  contentLabel="Remove?"
  isOpen={false}
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
>
  <h3
    className="modal__title"
  >
    Are you sure?
  </h3>
  <p
    className="modal__subtitle"
  >
    You will not be able to recover this expense.
  </p>
  <div
    className="button-group"
  >
    <button
      className="button button--secondary"
    >
      Cancel
    </button>
    <button
      className="button button--warning"
    >
      Yes, delete it!
    </button>
  </div>
</Modal>
`;
